hermeto/core/package_managers/pip.py

```diff
@@ -2231,6 +2232,41 @@ def _version(dep: dict[str, Any]) -> str:
     }
 
 
+def _has_rust_build_deps(raw_build_dependencies: list[str]) -> bool:
+    rust_build_deps = ("maturin", "setuptools-rust", "setuptools_rust")
+    for dep in raw_build_dependencies:
+        if dep.strip().lower().startswith(rust_build_deps):
+            return True
+    return False
+
+
+def _depends_on_rust(source_tarball: tarfile.TarFile) -> bool:
+    file_parser_map = {
+        "pyproject.toml": parsers.parse_pyproject_toml,
+        "setup.cfg": parsers.parse_setup_cfg,
+        "setup.py": parsers.parse_setup_py,
+    }
+    for file_name, parser in file_parser_map.items():
+        pkg_name = source_tarball.getnames()[0].split("/")[0]
+        try:
+            file = source_tarball.extractfile(f"{pkg_name}/{file_name}")
+        except KeyError:
+            continue
+        # the file is decoded as utf-8-sig because plain utf-8 has proven to
+        # be problematic with certain sources from pypi
+        # see https://github.com/hermetoproject/pybuild-deps/blob/4dc40ffabddb8aad1279978b8741111fb64452e6/src/pybuild_deps/finder.py#L45-L51
+        file_contents = file.read().decode("utf-8-sig") # type: ignore
+        try:
+            build_dependencies = parser(file_contents)
+        except parsers.SetupPyParsingError:
+            # unfortunately pybuild-deps parser has some known edge-cases for older packages
+            # relying on setup.py
+            log.error("Unable to parse build dependencies for %s.", pkg_name)
```

Comment:
You would probably want to continue to the next pair here or hide `if _has_rust_build_deps(build_dependencies):` in an `else:` clause: it looks like `build_dependencies` will be undefined when `log.error` is reached.

Reply:
nice catch! thanks